artpaint/application/UndoQueue.cpp

```diff
@@ -352,14 +352,14 @@ const char* UndoQueue::ReturnRedoEventName()
 
 void UndoQueue::UpdateMenuItems()
 {
-	const char *event_name;
+	BString event_name;
 	event_name = ReturnUndoEventName();
 
 	if (undo_menu_item != NULL) {
 		if (event_name != NULL) {
-			char menu_text[256];
-			sprintf(menu_text,"%s %s",B_TRANSLATE("Undo"),event_name);
-			undo_menu_item->SetLabel(menu_text);
+			BString menu_text;
+			menu_text.SetToFormat(B_TRANSLATE("Undo %s"), event_name);
```

Comment:
You either have to change to event_name.String(), or change to

```suggestion
			BString menu_text(B_TRANSLATE("Undo: %eventname%"));
			menu_text.ReplaceFirst("%eventname%", event_name);
```
Personally, I prefer the "ReplaceFirst()" route when dealing with string substitution and leave the SetToFormat() for dealing with non-string formats. Let's wait for @dsizzle to decide what he prefers.

In any case, I'd add a ":" after "Undo", because the event name starts with a capital letter and without a colon in front of it, it looks incorrect.

All the above applies to "Redo" below, too, ofc.

Reply:
Sorry, I won’t likely be able to look at this until Saturday.

Reply:
I am okay with both routes, just let me know what you guys prefer.